components/src/preact/locationFilter/location-filter.tsx

```diff
@@ -52,71 +49,179 @@ export const LocationFilterInner = ({ initialValue, fields, placeholderText }: L
     if (error) {
         throw error;
     }
+    if (data === undefined) {
+        return <NoDataDisplay />;
+    }
 
-    const onInput = (event: JSXInternal.TargetedInputEvent<HTMLInputElement>) => {
-        const inputValue = event.currentTarget.value;
-        setValue(inputValue);
-        if (inputValue.trim() === value.trim() && inputValue !== '') {
-            return;
-        }
-        const eventDetail = parseLocation(inputValue, fields);
-        if (hasAllUndefined(eventDetail) || hasMatchingEntry(data, eventDetail)) {
-            divRef.current?.dispatchEvent(
-                new CustomEvent('gs-location-changed', {
-                    detail: eventDetail,
-                    bubbles: true,
-                    composed: true,
-                }),
-            );
-            setUnknownLocation(false);
-        } else {
-            setUnknownLocation(true);
+    return <LocationSelector fields={fields} value={value} placeholderText={placeholderText} locationData={data} />;
+};
+
+type SelectItem = {
+    lapisFilter: LapisLocationFilter;
+    label: string;
+    description: string;
+};
+
+const LocationSelector = ({
+    fields,
+    value,
+    placeholderText,
+    locationData,
+}: LocationFilterInnerProps & {
+    locationData: LapisLocationFilter[];
+}) => {
+    const allItems = useMemo(
+        () =>
+            locationData
+                .map((locationFilter) => {
+                    return toSelectOption(locationFilter, fields);
+                })
+                .filter((item): item is SelectItem => item !== undefined),
+        [locationData, fields],
+    );
+
+    const initialSelectedItem = useMemo(
+        () => (value !== undefined ? toSelectOption(value, fields) : null),
+        [value, fields],
+    );
+
+    const [items, setItems] = useState(allItems.filter((item) => filterByInputValue(item, initialSelectedItem?.label)));
+    const divRef = useRef<HTMLDivElement>(null);
+
+    const shadowRoot = divRef.current?.shadowRoot ?? undefined;
+
+    const {
+        isOpen,
+        getToggleButtonProps,
+        getMenuProps,
+        getInputProps,
+        highlightedIndex,
+        getItemProps,
+        selectedItem,
+        inputValue,
+        selectItem,
+        setInputValue,
+    } = useCombobox({
+        onInputValueChange({ inputValue }) {
+            setItems(allItems.filter((item) => filterByInputValue(item, inputValue)));
+        },
+        onSelectedItemChange({ selectedItem }) {
+            if (selectedItem !== null) {
+                divRef.current?.dispatchEvent(new LocationChangedEvent(selectedItem.lapisFilter));
+            }
+        },
+        items,
+        itemToString(item) {
+            return item?.label ?? '';
+        },
+        initialSelectedItem,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        environment: shadowRoot,
+    });
+
+    const onInputBlur = () => {
+        if (inputValue === '') {
```

Comment:
```suggestion
    const onInputBlur = () => {
        closeMenu();
        if (inputValue === '') {
```

Where `closeMenu` is from the `useCombobox` return value. Then the autocompletion list closes when clicking somewhere outside (which seems to be the default behavior anyway when it's not used in a shadow DOM).